pkg/webhook/controlplane/ensurer.go

```diff
@@ -417,8 +417,8 @@ func (e *ensurer) EnsureKubeletServiceUnitOptions(ctx context.Context, gctx gcon
 
 func ensureKubeletCommandLineArgs(command []string, csiEnabled bool, kubeletVersion *semver.Version) []string {
 	if csiEnabled {
+		command = extensionswebhook.EnsureStringWithPrefix(command, "--cloud-provider=", "external")
 		if !version.ConstraintK8sGreaterEqual123.Check(kubeletVersion) {
-			command = extensionswebhook.EnsureStringWithPrefix(command, "--cloud-provider=", "external")
 			command = extensionswebhook.EnsureStringWithPrefix(command, "--enable-controller-attach-detach=", "true")
```

Comment:
Question that is not directly related to this change but rather for the K8s 1.23 support: why we do no longer set the `--enable-controller-attach-detach=true` flag for K8s 1.23. At least from a first look into the code and release notes I cannot spot a deprecation of this flag. Do you recall what was the reason?

Reply:
This setting is now set to `true` in the kubelet configuration file.

Reply:
Ah, I now see. Thanks for explaining!